docs/DeveloperGuide.md

```diff
@@ -1,34 +1,165 @@
+4. The command execution can affect the `Model` (e.g. clearing the list). 
+5. The result of the command execution is encapsulated as a `CommandOutput` object which is passed back to the `TextUi`.
+6. In addition, the `CommandOutput` object can also instruct the `TextUi` to perform certain actions, such as displaying the list to the user.
+
+Given below is the Sequence Diagram for interactions within the `Logic` component for the `parseCommand("clear)"` API call.
+
+![](images/LogicComponentSequenceDiagram.png?raw=true "Logic component Sequence Diagram for clear")
+*Figure #. Interactions Inside the Logic Component for the `clear` Command*
+
+### Model component
+
+**API** : [`seedu.duke.person`](https://github.com/AY2021S2-CS2113T-T09-1/tp/tree/master/src/main/java/seedu/duke/person) package
+
+The Model component,
+
+* contains a `TrackingList` class to keep track of all the `Person` objects.
+* contains a `Person` class which represents a person who checks in or out.
+
+A `Person` object contains:
+* an `Id` object
+* a `Name` object
+* a `Phone` object
+
+### Storage component
+**API** : [`seedu.duke.storage`](https://github.com/AY2021S2-CS2113T-T09-1/tp/tree/master/src/main/java/seedu/duke/storage) package
+![](images/StorageModule.png?raw=true "Storage Module diagram")
+
+The `Storage` component,
+* saves and encodes `Person` objects in `Tracking List` into a `.txt` file.
+* reads a `.txt` file of valid encoding and populates a `TrackingList`
+* keeps a list of user setting in a `Config.properties` file which is read on launch
+* defaults to `TrackingList.txt` but can be saved at a custom specified location
+* loads on program startup automatically saves after each command
+
+The storage module uses the `StorageFile` class to store the `TrackingList` class from Model. 
+`StorageFile` achieves this by using two helper classes, `TrackingListEncoder` and `TrackingListDecoder`.
+
+`TrackingListEncoder` takes the currently stored `TrackingList` and converts it to a more storage friendly String.
+This String is then written to a file on the disk, known on the diagram as `StoredTrackingList`.
+As the name suggests, `TrackingListDecoder` achieves the reverse by taking the file on drive, `StoredTrackingList`,
+and interprets the file to be loaded into the memory as a `TrackingList`.
+
+`StoredTrackingList` is the `.txt` file stored in the same directory as the program.
+The location is determined by the `ConfigFile` class. 
+
+The program settings for the user is manged by the `ConfigFile` class. 
+Another file, `settingsFile` is read by `ConfigFile` when the program is started, 
+to load the settings into the program memory. 
+When a change is made by the user, `ConfigFile` will update the `settingsFile` through a save function.
+
+### Common classes
+
+Classes used by multiple components are in the `seedu.Duke.commons` package. 
+
+## Implementation
```

Comment:
Perhaps there can be an implementation component to help developers understand how Safest Entry Tracker is implemented?